### PR TITLE
Add unit tests for ListControlStringCollectionEditor

### PR DESCRIPTION
**Pull request by Nora-Zhou01**
Related https://github.com/dotnet/winforms/issues/10773

## Proposed changes

- Add unit test ListControlStringCollectionEditorTests.cs for public properties and method of the ListControlStringCollectionEditor
- Enable nullability in ListControlStringCollectionEditorTests.cs

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12516)

### CODE REVIEW

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ListControlStringCollectionEditorTests.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+using System.ComponentModel;
+
+namespace System.Windows.Forms.Design.Tests;
+
+public class ListControlStringCollectionEditorTests
+{
+    [Fact]
+    public void EditValue_WithNullContext_ReturnsBaseEditValue()
+    {
+        var editor = new ListControlStringCollectionEditor(typeof(string));
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ListControlStringCollectionEditorTests.cs:16]:
```suggestion
        ListControlStringCollectionEditor editor = new(typeof(string));
```

Same in below tests as well

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ListControlStringCollectionEditorTests.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        object? result = editor.EditValue(context.Object, provider, value);
+
+        result.Should().Be(value);
+    }
+
+    [Fact]
+    public void EditValue_WithListControlAndNullDataSource_ReturnsBaseEditValue()
+    {
+        var editor = new ListControlStringCollectionEditor(typeof(string));
+        ListBox listControl = new();
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ListControlStringCollectionEditorTests.cs:43]: Please dispose ListBox
```suggestion
        using ListBox listControl = new();
```

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ListControlStringCollectionEditorTests.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+        ListBox listControl = new() { DataSource = new List<string> { "item1", "item2", "item3" } };
+
+        Mock<ITypeDescriptorContext> context = new();
+        context.Setup(c => c.Instance).Returns(listControl);
+
+        var provider = new Mock<IServiceProvider>().Object;
+        object? value = new();
+
+        ArgumentException exception = Assert.Throws<ArgumentException>(() => editor.EditValue(context.Object, provider, value));
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ListControlStringCollectionEditorTests.cs:67]:
```suggestion
        ArgumentException exception = ((Action)(() => editor.EditValue(context.Object, provider, value))).Should().Throw<ArgumentException>().Which;
```

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ListControlStringCollectionEditorTests.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
```
Comment (ricardobossan) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ListControlStringCollectionEditorTests.cs:3]: When trying to run the tests myself i could not build the code with errors:

```
The "Csc" task could not be initialized with its input parameters. 
The "InterceptorsNamespaces" parameter is not supported by the "Csc" task loaded from assembly: Microsoft.Build.Tasks.CodeAnalysis, Version=4.12.10.37901, Culture=neutral, PublicKeyToken=31bf3856ad364e35 from the path: C:\.tools\.nuget\packages\microsoft.net.sdk.compilers.toolset\9.0.100-preview.7.24407.12\Microsoft.Build.Tasks.CodeAnalysis.dll. Verify that the parameter exists on the task, the <UsingTask> points to the correct assembly, and it is a settable public instance property.
```

I think that is due to the commit from which you checked out from to make the tests. I solved this by rebasing with the main branch on the `runtime` repo remote. You can do this:

```sh 
gh pr checkout 12516 # If you are out of the branch now. I use github cli for this
git fetch --all
git pull --rebase $UPSTREAM_REMOTE main
git push --force
```

Comment (ricardobossan) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ListControlStringCollectionEditorTests.cs:3]: When trying to run the tests myself i could not build the code with errors:

```
The "Csc" task could not be initialized with its input parameters. 
The "InterceptorsNamespaces" parameter is not supported by the "Csc" task loaded from assembly: Microsoft.Build.Tasks.CodeAnalysis, Version=4.12.10.37901, Culture=neutral, PublicKeyToken=31bf3856ad364e35 from the path: C:\.tools\.nuget\packages\microsoft.net.sdk.compilers.toolset\9.0.100-preview.7.24407.12\Microsoft.Build.Tasks.CodeAnalysis.dll. Verify that the parameter exists on the task, the <UsingTask> points to the correct assembly, and it is a settable public instance property.
```

I think that is due to the commit from which you checked out from to make the tests. I solved this by rebasing with the main branch on the `runtime` repo remote. You can do this:

```sh 
gh pr checkout 12516 # If you are out of the branch now. I use github cli for this
git fetch --all
git pull --rebase $UPSTREAM_REMOTE main
git push --force
```

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ListControlStringCollectionEditorTests.cs [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+namespace System.Windows.Forms.Design.Tests;
+
+public class ListControlStringCollectionEditorTests
+{
+    [Fact]
+    public void EditValue_WithNullContext_ReturnsBaseEditValue()
+    {
+        var editor = new ListControlStringCollectionEditor(typeof(string));
+        var provider = new Mock<IServiceProvider>().Object;
```
Comment (ricardobossan) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ListControlStringCollectionEditorTests.cs:17]: Please declare variables with explicit types. This happens in other areas of the code as well.

Comment (ricardobossan) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ListControlStringCollectionEditorTests.cs:17]: Please declare variables with explicit types. This happens in other areas of the code as well.

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ListControlStringCollectionEditorTests.cs [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+        result.Should().Be(value);
+    }
+
+    [Fact]
+    public void EditValue_WithListControlAndNonNullDataSource_ThrowsArgumentException()
+    {
+        var editor = new ListControlStringCollectionEditor(typeof(string));
+
+        ListBox listControl = new() { DataSource = new List<string> { "item1", "item2", "item3" } };
```
Comment (ricardobossan) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ListControlStringCollectionEditorTests.cs:59]: Please dispose of this control with `using`

Comment (ricardobossan) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ListControlStringCollectionEditorTests.cs:59]: Please dispose of this control with `using`

**Review by ricardobossan — COMMENTED** [review]
Other than small comments, LGTM. Just make sure to rebase with updated main remote upstream branch, to get rid of the errors when trying to build.

**Review by lonitra — APPROVED** [review]

**Pull request merged**